src/Client.cc

```diff
@@ -40,14 +41,43 @@ static const std::string CFG_TLS_ALLOW_INSECURE = "tlsAllowInsecureConnection";
 static const std::string CFG_STATS_INTERVAL = "statsIntervalInSeconds";
 static const std::string CFG_LOG = "log";
 
+LogCallback *Client::logCallback = nullptr;
+
+void Client::SetLogHandler(const Napi::CallbackInfo &info) {
+  Napi::Env env = info.Env();
+  Napi::HandleScope scope(env);
+  Napi::Value jsFunction = info[0];
+
+  if (jsFunction.IsNull()) {
+    if (Client::logCallback != nullptr) {
+      Client::logCallback->callback.Release();
+      delete Client::logCallback;
+      Client::logCallback = nullptr;
+    }
+  } else if (jsFunction.IsFunction()) {
+    Napi::ThreadSafeFunction logFunction =
+        Napi::ThreadSafeFunction::New(env, jsFunction.As<Napi::Function>(), "Pulsar Logging", 0, 1);
+    logFunction.Unref(env);
```

Comment:
[question] What is the reason for doing `Unref` here?

Reply:
I found that setting the global log function without unrefing it keeps the process alive. This is to allow the process to exit cleanly.

Reply:
Thanks.